### PR TITLE
Always use builtins for floor/ceil/trunc/fabs/copysign/sqrt

### PR DESCRIPTION
**Pull request by quantum5**
Refs #9236.

### CODE REVIEW

File: system/lib/libc/musl/src/math/ceil.c [hunk 1]
```diff
@@ -28,4 +33,5 @@ double ceil(double x)
 	if (y < 0)
 		return x + y + 1;
 	return x + y;
+#endif
```
Comment (sbc100) [inline system/lib/libc/musl/src/math/ceil.c]: This seems little odd to me.   Surely the compiler will normally convert `ceil` to `__builtin_ceil` at the callsite?

So this change only effect code built explicitly with `-fno-builtin`?    i.e it effect the libc itself, not moew user code?

It seems like there should be more precise way to compile libc that building the whole thing with `-fno-builtin`.

File: tools/system_libs.py [hunk 2]
```diff
@@ -665,7 +664,8 @@ def get_files(self):
       # With the wasm backend these are included in wasm_libc_rt instead
       blacklist += [os.path.basename(f) for f in get_wasm_libc_rt_files()]
     else:
-      blacklist += ['rintf.c']
+      blacklist += ['rintf.c', 'ceil.c', 'ceilf.c', 'floor.c', 'floorf.c',
+                    'fabs.c', 'fabsf.c', 'sqrt.c', 'sqrtf.c']
```
Comment (sbc100) [inline tools/system_libs.py]: Is there a reason these still need blacklisting for asmjs?

Comment (quantum5) [reply tools/system_libs.py]: Blacklisting these will use the JS implementations, which makes sense for asm.js.

Comment (sbc100) [reply tools/system_libs.py]: Won't the clang builtins also to the right thing for asmjs?

Comment (sbc100) [reply tools/system_libs.py]: Won't the clang builtins also to the right thing for asmjs?

Comment (quantum5) [reply tools/system_libs.py]: From what I heard, asmjs is supposed to just use `Math.floor` etc. directly. @kripken would know more.

Comment (kripken) [reply tools/system_libs.py]: I think those builtins were added for upstream. I don't think fastcomp has them all. It relies on JS library stuff to handle those without intrinsics.

**Comment by dschuff** [comment]
FWIW I like their implementation (just put the functions in a different file) better than ours, since it doesn't involve hacking the upstream musl files.
I'm kind of surprised musl doesn't have an existing way of doing this; surely there must be other targets that want to use e.g. assembly code instead of the default implementation for some of these?

**Pull request merged**